### DISCUSSION TITLE
cheat idea: negative money

### DISCUSSION THREAD

**Discussion by Lipti-pl** [winstxnhdw/lc-hax#546] [General]
the game is written in such a way that makes it so when your money goes into negatives you cant physically change the moons without external mods 
i think this would be a great cheat for fucking lobbies over o7
this could be performed by using the same code behind the /buy command except you change the price tag from well 1 to a very ludicrous number

**Comment by winstxnhdw** [comment]
Commands that change credits are host-only.

**Reply by Lipti-pl** [reply]
> Commands that change credits are host-only.

nono you understood me wrong 
when youre doing /buy you spend 1 credit out of the total credits that the ship terminal has 
you could keep /buy the same and make another command like idfk /inflation that either does the same thing an insane ammount of time or just buys one item for a lot of credits so it goes into negatives

it already works on client side and you dont have to be a host to do this with /buy command

**Reply by winstxnhdw** [reply]
IIRC, there’s a check on the host’s side that will verify if you can afford the item. So this wouldn’t work.

**Comment by winstxnhdw** [comment]
But fwiw, I have removed the clamping on `/credit`

**Comment by Lipti-pl** [comment]
How come ive made the money count go into negatives multiple  times by doing /buy then?   I was on client side not host and people complained about not being able to switch planets due to well negative money  I fully believe you with the check idea but what happened in my lobbies then XD 
     
      
       
        Dnia 16 września 2024 17:06 Winston H. ***@***.***&gt; napisał(a):
       
    
       
         
 IIRC, there’s a check on the host’s side that will verify if you can afford the item. So this wouldn’t work.

**Reply by winstxnhdw** [reply]
Seems like they are unable to switch planets because there’s insufficient money. Not because the money is in the negatives.

**Comment by Lipti-pl** [comment]
Me when we cant go to Gordon or any free planet when i do that but alright  
     
      
       
        Dnia 18 września 2024 12:21 Winston H. ***@***.***&gt; napisał(a):
       
    
       
         
 Seems like they are unable to switch planets because there’s insufficient money. Not because the money is in the negatives.

**Comment by winstxnhdw** [comment]
Idh time to investigate this atm. You can try asking @xAstroBoy

**Reply by xAstroBoy** [reply]
Like you said, money is host only, and i think setting it as a negative, it will break the game purchasing mechanism, i think this is a waste of time .

**Reply by ghost** [reply]
It seems that the issue could be resolved by changing `terminal.groupCredits - 1` to 0 in the `BuyCommand.cs` file.

**Reply by winstxnhdw** [reply]
Elaborate? What does this achieve other than making your credits 0?

**Reply by ghost** [reply]
When using the /buy command to make a purchase, 1 dollar is deducted from the credit. However, if this amount is set to 0, the actual money is not deducted. I tested this by entering the number 1000, and I confirmed that 1000 credits were deducted, even though I was not the host.

**Reply by winstxnhdw** [reply]
That’s what we used to do, but IIRC, there was an update in Lethal Company that ensured your new credit is less than your original credit.

Sometimes the credit is incorrectly reflected on only your end. If you have a friend to play with, you could ask them to see if the credit has indeed remained the same on their end. You should also make sure that the item actually comes.

**Reply by xAstroBoy** [reply]
i still think this is a useless edit and only a way to ruin a game.

**Reply by ghost** [reply]
and When using the "hear" command, I noticed that the entire voice was audible, but the notifications were not working. This was because both the buyer and the deceased person could hear each other, leading everyone to feel that they were all still alive. To fix this, I added the code `if (!player.isPlayerDead) return;` in the HearPatch.cs file. After this change, I confirmed that both the voice and notifications worked properly.

**Reply by winstxnhdw** [reply]
> i still think this is a useless edit and only a way to ruin a game.

If we can get true free purchases again, it’d be pretty useful though.

**Reply by winstxnhdw** [reply]
> This was because both the buyer and the deceased person could hear each other

What does this even mean?

**Reply by ghost** [reply]
Yes, the same fix applied to others as well. However, I noticed that when a person dies and the ship rises, the negative credits are reset to 0.

**Reply by winstxnhdw** [reply]
> To fix this, I added the code `if (!player.isPlayerDead) return;` in the HearPatch.cs file.

This doesn't sound right. This means you are ignoring all living players. That sounds like it defeats the purpose of using `hear`.

**Reply by ghost** [reply]
> > 이는 구매자와 사망자가 서로의 목소리를 들을 수 있었기 때문이다.
> 
> 이게 무슨 뜻이지?

It's because you can hear everyone's voice.  *

**Reply by winstxnhdw** [reply]
Yes, that is what `hear` does. It's meant for you to hear everyone.

**Reply by ghost** [reply]
If that's the case, the Notification feature doesn't work.

**Reply by ghost** [reply]
I don't want to give up on either of them T-T.

**Reply by winstxnhdw** [reply]
Honestly, I don't understand why `hear` is breaking notifications. I'll take a look but I can't guarantee that I'll be able to figure this one out.

**Reply by ghost** [reply]
When you can hear everyone's voice, isn't it possible that it leads to the conclusion that there are no deceased players? That's why it seems like it doesn't show as "dead."

**Reply by ghost** [reply]
When I used `if (!player.isPlayerDead) return;`, I was unable to hear everyone's voice, but I could hear the voices of the deceased players.

**Reply by winstxnhdw** [reply]
When a player "dies", they are sent to `/void`. Their speaker object remains in the same world as you. Since `hear` allows you to hear any players infinitely far away, you would also be able to hear dead players.

**Reply by winstxnhdw** [reply]
The point of `hear` is to hear everyone. Not just dead players. I could update the command to only hear the dead but I am not sure if this is a feature anyone wants.

**Reply by ghost** [reply]
Yes, that's right. I want to hear everyone's voice, but thanks to the amazing program you've created, it's hard to give up on anything. T-T

**Reply by ghost** [reply]
Thank you for creating such a great file

**Reply by winstxnhdw** [reply]
No worries. Like I said, I’ll try looking into the issue but I can’t promise anything. `hear` has always been a buggy mess because DissonanceVoIP is garbage.
